### DISCUSSION TITLE
Issues with prefix/suffix in EssX commands

### DISCUSSION THREAD

**Discussion by Chris6ix** [EssentialsX/Essentials#5133] [Q&A] [answered]
I haven't played minecraft for a while now, and I just updated my private server and played a bit with 2 friends, and I have 2 issues that I have no idea how to fix/how they even happened...

I have set my chat format in EssX set to `{SUFFIX} {PREFIX} {USERNAME}` and it works in normal chat messages, but when I do `/balancetop` it looks like this: `{PREFIX} {USERNAME}{SUFFIX}`.
I also have the issue that EssX commands don't work with tab completition anymore because it puts the prefix and suffix into the tab completition...(E.g. when I type `/tpa` and then press tab, I see all the players listed as `{PREFIX} {USERNAME}{SUFFIX}`, just like in balancetop)
I know that this wasn't like that when I last played (Which is quite a while ago tho...)

If it helps, I'm using `EssentialsX-2.19.7` and `EssentialsXChat-2.19.7` on `paper-1.19.2-210`

If anyone could maybe tell me what causes this issue or how I can fix it, that would be amazing :)

**Comment by pop4959** [comment, marked as answer]
Suffix is intended to be placed after the username (to the definition of the word). Anything before the name should be part of the prefix.

**Reply by Chris6ix** [reply]
Hmm...but why did it work before then?
Because I use the suffix for "titles" so that it looks for example like `Veteran VIP Chris6ix` and that has worked perfectly for quite a long time...
And it seems like this is an issue only happening with EssentialsX commands...all the commands from the other plugins that can target players are fine...

Is there any way to force EssX to use `{SUFFIX} {PREFIX} {USERNAME}`?
And how do I stop the prefix and suffix from showing up in tab completition for EssX commands?

**Comment by mdcfe** [comment]
EssentialsX has never used the chat format to determine the order of placeholders in a player's display name. The chat format *only* controls the order that placeholders appear in public chat messages.

If you're looking for the ability to combine different titles, I'd look into [LuckPerms' prefix and suffix stacking](https://luckperms.net/wiki/Prefix-&-Suffix-Stacking). This will let you combine prefixes from multiple ranks into a single prefix, which EssentialsX and other plugins can then access through Vault.

**Reply by Chris6ix** [reply]
Okay, I took a look at this and have now fixed the title being behind the name in balancetop, but I still have the issue that the prefix is showing up in the tab competition of EssX commands...how can I fix that?

**Reply by Chris6ix** [reply]
@mdcfe ?

**Reply by Evidentsinger14** [reply]
> I still have the issue that the prefix is showing up in the tab competition of EssX commands...how can I fix that?

There is a config setting to enable/disable it. If you search for `tab-complete` you should find it.

**Reply by Chris6ix** [reply]
Oh...I see
Thank you very much @Evidentsinger14 :)
I must have missed that setting when I created a new config while updating EssX...

Also thanks to mdcfe for that other fix :)

**Reply by Evidentsinger14** [reply]
It’s not enabled by default, that would have been modified if you generated a new config - glad to help though! 🙂
